components/brave_rewards/resources/ui/components/checkout/creditCardForm/index.tsx

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import { Toggle } from 'brave-ui/components'
+
+import { Behaviors } from './behaviors'
+import * as Icons from './icons'
+import { CreditCardDetails, CreditCardError } from './types'
+
+import {
+  Container,
+  CardNumber,
+  Expiration,
+  SecurityCode,
+  SaveThisCard,
+  InputIconContainer
+} from './style'
+
+export { CreditCardDetails }
+
+export interface CreditCardFormHandle {
+  focus: () => void
+  validate: () => Array<CreditCardError>
+  details: CreditCardDetails
+}
+
+interface CreditCardFormProps {
+  handleRef: React.MutableRefObject<CreditCardFormHandle | null>
+}
+
+export function CreditCardForm (props: CreditCardFormProps) {
+  const [saveCardChecked, setSaveCardChecked] = React.useState(true)
+  const toggleSaveCard = () => { setSaveCardChecked(!saveCardChecked) }
+
+  const behaviorsRef = React.useRef<Behaviors | null>(null)
+  const cardNumberRef = React.useRef<HTMLInputElement>(null)
+  const expiryRef = React.useRef<HTMLInputElement>(null)
+  const securityCodeRef = React.useRef<HTMLInputElement>(null)
+
+  React.useEffect(() => {
+    let behaviors = behaviorsRef.current
+    if (!behaviors) {
+      behaviorsRef.current = behaviors = new Behaviors({
+        inputs: {
+          cardNumber: cardNumberRef.current!,
+          expiry: expiryRef.current!,
+          securityCode: securityCodeRef.current!
+        }
+      })
+    }
+    props.handleRef.current = behaviors
+  })
+
+  return (
+    <Container>
+      <CardNumber>
+        <label>
+          Card number
+          <input ref={cardNumberRef} autoComplete='cc-number' />
```

Comment:
can we use input from brave-ui as it already has handler for icons?

Reply:
Unfortunately, `<Input />` doesn't currently expose a `ref` prop and it doesn't accept things like `autoComplete`. https://github.com/brave/brave-ui/issues/373